### PR TITLE
Add seller registration

### LINKED ISSUE #29: Add Seller Registration
According to the rules of the task, users should be able to register as sellers. 

**Right now we have the following roles:**

- `user`, does not have any scopes. Can only register and see products 

- `admin`, has scopes `users products:write media:write`
   - `users` gives permissions to get, create and update users `using /users/...`
   - `products:write` allows to create and update your own products
   - `media:write` allows uploading media

### To be implemented

- [x] A new role `seller` should be added. The `seller` should have the following scopes:
   - `products:write` 
   - `media:write`

- [x] The registration of the `seller` should be implemented using the existing `/auth/register` endpoint by adding a `role` field. 

- [x] `admin` scopes should be changed 
   - `users` will be replaced by `users:admin`
   - `products:write` will be replaced by `products:admin`. Admins will be able to change all products on the platform

### PR DESCRIPTION
**Pull request by maximihajlov**
Resolves #29. Adds a new role `seller`, who can create and update products

### CODE REVIEW

File: backend/services/users/src/main/java/pw/mer/letsplay/web/AuthController.java [hunk 1]
```diff
@@ -102,7 +105,23 @@ public String register(@Valid @RequestBody RegisterRequest request) {
...
+                throw new ResponseStatusException(BAD_REQUEST);
+            }
+            if (role == ERole.ADMIN) {
+                throw new ResponseStatusException(BAD_REQUEST, "Can not register as admin");
+            }
+        } else {
+            role = ERole.USER;
+        }
+
+        var user = new User(request.name, request.email, passwordEncoder.encode(request.password), role);
```
[the remark this answers is not in the document]
Comment (adrian-lui) [reply backend/services/users/src/main/java/pw/mer/letsplay/web/AuthController.java:124]: resolved in #33

File: backend/services/users/src/main/java/pw/mer/letsplay/web/UserController.java [hunk 2]
```diff
@@ -40,9 +40,21 @@ public List<User> index() {
...
                 new ResponseStatusException(NOT_FOUND, NOT_FOUND_MESSAGE));
+
+        if (authentication != null && authentication
+                .getAuthorities()
+                .contains(new SimpleGrantedAuthority("SCOPE_users:admin"))
+        ) {
+            return user;
+        }
+
+        if (user.getRole() != ERole.SELLER) {
```
[the remark this answers is not in the document]
Comment (adrian-lui) [reply backend/services/users/src/main/java/pw/mer/letsplay/web/UserController.java:54]: resolved in #33

File: backend/services/products/src/test/java/pw/mer/letsplay/AbstractControllerTests.java [hunk 3]
```diff
@@ -12,8 +11,12 @@ public abstract class AbstractControllerTests extends SharedAbstractControllerTe
...
     private ProductRepo productRepo;
 
-    @Override
     public String getAdminToken() {
+        var admin = new SharedAuthFactory.TestUser(List.of("products:admin"));
+        return getAccessToken(admin);
+    }
+
+    public String getSellerToken() {
         var admin = new SharedAuthFactory.TestUser(List.of("products:write"));
```
[the remark this answers is not in the document]
Comment (adrian-lui) [reply backend/services/products/src/test/java/pw/mer/letsplay/AbstractControllerTests.java:20]: resolved in #32 #33

**Review by adrian-lui — APPROVED** [review]
Approved. comments are fixed in #32 #33

**Comment by maximihajlov** [comment]
Thanks for the review 🚀

**Pull request merged**